### ISSUE TITLE
Shaded boxes around links

### ISSUE THREAD

**Issue by Fontanapink** [latex3/hyperref#120]
Instead of having hyperref color links with a box or changing the color of the text, it would be awesome to have a shaded area around links like this:

![image](https://user-images.githubusercontent.com/12236668/74045483-9b2eeb00-49ab-11ea-825d-280fb52b9694.png)

**Comment by u-fischer** [comment]
the pdf specification doesn't knows an easy way to do this. One could use a \colorbox but this would disable line breaks in the link and also prevent spaces to stretch. An appeareance dictionary wouldn't work in all pdfviewers and would prevent line breaks too.

**Comment by dbitouze** [comment]
> One could use a `\colorbox` but this would disable line breaks in the link

@u-fischer `soul`s `\hl` wouldn't have this problem:

```latex
\documentclass{article}
\usepackage{xcolor}
\usepackage{soul}
\begin{document}
\hl{test test test test test test test test test test test test test test test
  test test test test test test test test test test test test test}
\end{document}
```
![test](https://user-images.githubusercontent.com/1032633/74060066-7bb6b300-49e9-11ea-88bd-0f6b84afcb95.png)

**Comment by u-fischer** [comment]
@dbitouze yes you can use it e.g in \href. But I wouldn't want to try to combine this e.g. with \url.

**Comment by Fontanapink** [comment]
Exactly. I need links to be highlighted like that. Better yet if shaded instead of highlighted.
